syntax = "proto3";

import "corpc_option.proto";

option cc_generic_services = true;

package wukong.pb;

message ForwardInRequest {
    int32 type = 1; // 消息类型
    uint32 tag = 2; // 消息tag（用于客户端与服务器RPC）
    repeated uint32 ids = 3; // 发送目标（角色号列表）
    bytes rawMsg = 4; // 原始消息数据
}

message EnterGameRequest {
    uint32 roleId = 1;
    uint32 lToken = 2;
    uint32 gatewayId = 3;
}

// 玩家游戏对象所在服务器需要实现GameService服务
service GameService {
    option (corpc.global_service_id) = 2;

    rpc forwardIn(ForwardInRequest) returns(corpc.Void)
    {
        option (corpc.not_care_response) = true;
    }

    // 注意：enterGame方法用于告诉游戏服向客户端发角色初始化数据，该方法可以重复调用，不应改变数据
    rpc enterGame(EnterGameRequest) returns(corpc.Void)
    {
        option (corpc.not_care_response) = true;
    }
}